src/database/models/user_item.rs

```diff
@@ -371,32 +352,26 @@ impl User {
 
     pub async fn clear_caches(
         user_ids: &[(UserId, Option<String>)],
-        redis: &deadpool_redis::Pool,
+        redis: &RedisPool,
     ) -> Result<(), DatabaseError> {
-        let mut redis = redis.get().await?;
-        let mut cmd = cmd("DEL");
-
         for (id, username) in user_ids {
-            cmd.arg(format!("{}:{}", USERS_NAMESPACE, id.0));
+            redis.delete(USERS_NAMESPACE, id.0).await?;
```

Comment:
This redis delete was previously a single operation, now it's `user_ids.len()` operations. I think we want the batching wherever we can get it.

Maybe we could hava `delete_many` method on the `RedisPool` that accepts a slice of tuples of something like `(String, impl Display)`? Though in this case we actually have two types - UserIds and Option<String>... so maybe it should actually take an `impl IntoIterator<Item = (String, String)>`?

Reply:
Looks like the clear_cache methods that use `delete_many` now still aren't as batched as they were - many are still calling `redis.delete_many` inside of their for loops. I think instead they should be building an iterator of the keys to delete, then calling `delete_many` once at the end.

Also it doesn't appear to be used in `version_item.rs`, `user_item.rs` are batched at all

Reply:
Looks good!